Core/InteropServices/GenericSafeHandle.cs

```diff
@@ -25,7 +25,11 @@ public class GenericSafeHandle : SafeHandleZeroOrMinusOneIsInvalid
 		/// <param name="closeMethod">The delegate method for closing the handle.</param>
 		/// <param name="ownsHandle"><see langword="true"/> to reliably release the handle during the finalization phase; <see langword="false"/> to prevent reliable release (not recommended).</param>
 		/// <exception cref="System.ArgumentNullException">closeMethod</exception>
-		public GenericSafeHandle(IntPtr ptr, Func<IntPtr, bool> closeMethod, bool ownsHandle = true) : base(ownsHandle) => SetHandle(ptr);
+		public GenericSafeHandle(IntPtr ptr, Func<IntPtr, bool> closeMethod, bool ownsHandle = true) : base(ownsHandle)
+		{
+			SetHandle(ptr);
+			CloseMethod = closeMethod;
```

Comment:
I was looking at a previous version of this and it appears I broke it back in 2018 :( Before I broke it, I also handled a null param as follows. Will you please add this and resubmit?
`CloseMethod = closeMethod ?? throw new ArgumentNullException(nameof(closeMethod));`

Reply:
Ok.
Personally I disagree with using `?? throw new ArgumentNullException(nameof(closeMethod))` because it means that you do much work just to find out that you didn't need to do it.

Reply:
Pushed an update